float32 width
---
int8 success